ipld/ipldsch/routing.ipldsch

```diff
@@ -0,0 +1,51 @@
+
+type ServiceEnvelope union {
+     | GetP2PProvideRequest "get-p2p-provide-request"
+     | GetP2PProvideResponse "get-p2p-provide-response"
+} representation keyed
+
+# TODO: IPLD schema type aliases not implemented
+# type Multihash = Bytes
+type Multihash struct {
+     Bytes Bytes
+}
+
+type GetP2PProvideRequest struct {
+     Keys [Multihash]
+}
+
+type ProvidersByKey struct {
+	Key Multihash
+	Provider Provider
+}
+
+type GetP2PProvideResponse struct {
+     # NOTE: ProvidersByKey {Multihash : Provider} is not possible in IPLD schema, instead:
+	ProvidersByKey [ProvidersByKey]
+}
+
+type Provider struct {
+     Node [Node]
+     Proto [TransferProtocol]
+     # can include other provider record meta here, e.g. the provenance of the provider record itself
+}
+
+# Node describes a node (identity, network address, authentication, whatever else applies)
+# We expect different types of nodes, e.g. peer, miner, public IP, etc.
+type Node union {
+     | Peer "peer"
+} representation keyed
+
+type Peer struct {
+     ID Bytes
+     Multiaddresses [Bytes]
+}
+
+type TransferProtocol union {
+     | BitswapTransfer "bitswap"
```

Comment:
I remember discussions about if a string namespace is what we want for this versus having these potential protocols represented with a numeric registry.

Our current ingest library goes with the numeric registery approach, with
`0x3E0000` for [bitswap](https://github.com/filecoin-project/index-provider/blob/main/metadata/bitswap.go#L9)
and
`0x3F0000` for [filecoin graphsync](https://github.com/filecoin-project/index-provider/blob/main/metadata/metadata.go#L63)

I think the reason we chose numeric namespace is that we expected the record formats may evolve, so even if a later record is also 'bitswap', it might be a 'bitswap v1.1+extra metadata' record of some sort, and identification of both protocol and record format seemed like it could become unwieldy with strings.

I don't have a strong preference either way here.

Reply:
These two have been now canonicalized in the multicodec table as `0x900` and `0x910` respectively